diff --git a/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake b/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
index ec0950f5e4d..8a8f1572e58 100644
--- a/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/flatbuffers.cmake
@@ -23,11 +23,11 @@ OverridableFetchContent_Declare(
   flatbuffers
   GIT_REPOSITORY https://github.com/google/flatbuffers
   # Keep in sync with tensorflow/third_party/flatbuffers/workspace.bzl
-  GIT_TAG v25.9.23
-  # NOTE: b/340264458 - `GIT_SHALLOW TRUE` works for tag name only,
+  GIT_TAG 187240970746d00bbd26b0f5873ed54d2477f9f3
+  # NOTE: b/340264458 - `GIT_SHALLOW FALSE` works for tag name only,
   # so if you change the `GIT_TAG` above to a hash rather than a tag name,
   # you need to also change the line below to `GIT_SHALLOW FALSE`.
-  GIT_SHALLOW TRUE
+  GIT_SHALLOW FALSE
   GIT_PROGRESS TRUE
   SOURCE_DIR "${CMAKE_BINARY_DIR}/flatbuffers"
 )
